components/server/src/workspace/workspace-service.ts

```diff
@@ -456,7 +467,7 @@ export class WorkspaceService {
             };
         }
 
-        if (workspace.type !== "regular") {
+        if (restrictToRegular && workspace.type !== "regular") {
```

Comment:
I wonder whether here we can check whether it is first start or restart instead? and only throw if we are trying to restart a non regular workspace?

Reply:
Note that the API does not allow to start any non-regular workspace. And we already do have a check for "is this a restart?" in WorkspaceStarter.startWorkspace. I agree that we eventually should move that check up here.

For now, I opted to keep the existing behavior as close to GitpodServerImpl as possible (especially in my current context, where I did not intend to touch it originally).